common/screenshotter.go

```diff
@@ -215,14 +216,9 @@ func (s *screenshotter) screenshot(
 	}
 
 	// Save screenshot capture to file
-	// TODO: we should not write to disk here but put it on some queue for async disk writes
 	if path != "" {
-		dir := filepath.Dir(path)
-		if err := os.MkdirAll(dir, 0o755); err != nil {
-			return nil, fmt.Errorf("creating screenshot directory %q: %w", dir, err)
-		}
-		if err := os.WriteFile(path, buf, 0o644); err != nil {
-			return nil, fmt.Errorf("saving screenshot to %q: %w", path, err)
+		if err := s.persister.Persist(path, bytes.NewBuffer(buf)); err != nil {
```

Comment:
I suggest wrapping this within a `bufio.Reader` for better performance, especially important for large files.

Reply:
Yeah, i was wondering the same, and I was going to add it to the `LocalPersister.Persist` method itself (it feels like that should decide the best course of action on how to buffer things). Actually maybe it's better to have a `bufio.Writer` since the screenshot bytes are already in memory, so we want to buffer the write to the disk.

Reply:
Added it to 897862ba11263a6edc1793f03c73892beaa2dbd1. Actually it probably doesn't matter if it's a `bufio.Reader` or `bufio.Writer` as long as it writes a small amount to disk to avoid slowing down the app with large IO.

Reply:
👍 Using `bufio.Writer` in the `Copy` makes sense.